ldk/goja/src/network/index.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * The Network Aptitude provides access to network calls through Olive Helps
+ */
+export interface Network {
+  /**
+   * Generates a HTTP request with the provided configuration.
+   *
+   * @param request - The HTTP request to make.
+   * @returns A Promise resolving with the response.
+   */
+  httpRequest(req: OliveHelps.HTTPRequest): Promise<OliveHelps.HTTPResponse>;
+}
+
+/**
+ * The HTTP Request configuration.
+ */
+export interface HTTPRequest {
+  body: Uint8Array;
+  headers: Record<string, string[]>;
+  method: string;
+  url: string;
+}
+
+/**
+ * The HTTP Response data.
+ */
+export interface HTTPResponse {
+  statusCode: number;
+  /**
+   * The HTTP response as a byte array. To decode into a UTF-8 string you can:
+   * ```
+   * import * as util from 'util';
+   * ...
+   * var string = new TextDecoder("utf-8").decode(uint8array);
+   * ```
+   */
+  data: Uint8Array;
+  headers: Record<string, string[]>;
+}
+
+function httpRequest(req: OliveHelps.HTTPRequest): Promise<OliveHelps.HTTPResponse> {
```

Comment:
Not sure if it is just preference, but should our method naming convention be different from parameters. For example function request(httpRequest: HTTPRequest). Also since you exporting 'HTTPResponse' as part of the network/index.ts you probably should return it instead of the OliveHelps.HTTPResponse. If our external contract changes we could remap it.

Reply:
You're right, I can change that really to use the locally defined interface